rosidl_runtime_py/convert.py

```diff
@@ -176,15 +176,13 @@ def message_to_ordereddict(
     """
     d = OrderedDict()
 
-    # We rely on __slots__ retaining the order of the fields in the .msg file.
-    for field_name, field_type in zip(msg.__slots__, msg.SLOT_TYPES):
+    # We rely on the get_fields_and_field_types() method to retrieve the fields in the .msg file.
```

Comment:
This comment is now a lot more important to explain what is going on.  So I think we should update it to something like:

```suggestion
    # The get_fields_and_field_types() method returns a map of field_names to stringified versions of the field types.
    # But here we want the "rosidl_parser.definition" types, so we zip the field names together with SLOT_TYPES.
    # The length of these two is guaranteed to be the same length by the Python code generator.
```

(and we should put the same comment up above)

Reply:
Ok makes sense. Done.